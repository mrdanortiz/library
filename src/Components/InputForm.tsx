import React, { FunctionComponent } from "react";
import { Form } from "semantic-ui-react";

interface Props {
    setFormData: (name: string, value: string | boolean) => void;
    getImages: () => void;
}

export const InputForm: FunctionComponent<Props> = ({ setFormData, getImages }) => (
    <Form>
        <Form.Input name="inputBox" onChange={(e, { name, value }) => setFormData(name as string, value as string)} />
        <Form.Checkbox
            name="checkbox"
            label="Find a single image"
            onChange={(e, { name, checked }) => setFormData(name as string, checked as boolean)}
        />
        <Form.Button type="submit" onClick={() => getImages()}>
            Search
        </Form.Button>
    </Form>
);

export default InputForm;
