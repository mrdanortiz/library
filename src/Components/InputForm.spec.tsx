import React from "react";
import renderer from "react-test-renderer";
import { InputForm } from "./InputForm";

const setFormData = jest.fn();
const getImages = jest.fn();

it("renders correctly", () => {
    const tree = renderer.create(<InputForm setFormData={setFormData} getImages={getImages} />).toJSON();
    expect(tree).toMatchSnapshot();
});
