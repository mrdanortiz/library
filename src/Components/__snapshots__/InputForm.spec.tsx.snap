// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <div
      className="ui input"
    >
      <input
        name="inputBox"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <div
      className="ui checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        name="checkbox"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label>
        Find a single image
      </label>
    </div>
  </div>
  <div
    className="field"
  >
    <button
      className="ui button"
      onClick={[Function]}
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
